org.eclipse.eclemma.core/src/org/eclipse/eclemma/internal/core/launching/AdjustedLaunchConfigurationWorkingCopy.java

```diff
@@ -55,6 +58,29 @@ public void setAttribute(String attributeName,
     throw new UnsupportedOperationException();
   }
 
+  public void setAttribute(String attributeName, String value) {
+    if (VM_ARGUMENTS_KEY.equals(attributeName)) {
+      delegate.setAttribute(attributeName, filterExtraVMArguments(value));
+    } else {
+      delegate.setAttribute(attributeName, value);
+    }
+  }
+
+  private String filterExtraVMArguments(String vmArguments) {
+    return vmArguments.substring(extraVMArgument.length()).trim();
```

Comment:
This implementation assumes, that the extra argument is still at the beginning of the string. If extra values are prepended this will break. I would therefore go with the following implementation:

     return vmArguments.replace(extraVMArgument, "").trim(); //$NON-NLS-1$

Reply:
This was actually my first implementation. But then I thought - What if we have the same argument in extraVMArguments and in VM arguments attribute - we will remove both arguments and modify the underlying value of the VM arguments attribute. This was only a hypothetical scenario, and I'm not sure if it's even possible. If you know it's not possible then I will change it as you suggested. Maybe a good compromise would be removing only the first occurrence like:
`return vmArguments.replaceFirst(Pattern.quote(extraVMArgument), "").trim(); //$NON-NLS-1$`
What do you think?

Reply:
If there is a JaCoCo agent with the same config this will definitely not work anyways. The only purpose of this class is to add the JaCoCo agent.

Reply:
OK. So let me replace it with your suggestion.